tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/OverlayAccess.java

```diff
@@ -0,0 +1,327 @@
+					ToolItem buttonForSearchOption= getButtonForSearchOption(option);
+					if (option == SearchOptions.GLOBAL) {
+						return buttonForSearchOption.getEnabled();// The "Global" option is mapped to a button that
+						// selects whether to search in the selection, thus inverting the semantic
+					}
+					return buttonForSearchOption != null && buttonForSearchOption.getEnabled();
+				})
+				.collect(Collectors.toSet());
+	}
+
+	private Set<SearchOptions> getSelectedOptions() {
+		return Arrays.stream(SearchOptions.values())
+				.filter(isOptionButtonSelected())
+				.collect(Collectors.toSet());
+	}
+
+	private Predicate<? super SearchOptions> isOptionButtonSelected() {
+		return option -> {
+			ToolItem buttonForSearchOption= getButtonForSearchOption(option);
+			if (option == SearchOptions.GLOBAL) {
+				return !buttonForSearchOption.getSelection();// The "Global" option is mapped to a button that
+				// selects whether to search in the selection, thus inverting the semantic
+			}
+			return buttonForSearchOption != null && buttonForSearchOption.getSelection();
+		};
+	}
+
+	public void pressSearch(boolean forward) {
+		if (forward) {
+			searchForward.notifyListeners(SWT.Selection, null);
+		} else {
+			searchBackward.notifyListeners(SWT.Selection, null);
+		}
+	}
+
+	@Override
+	public IFindReplaceLogic getFindReplaceLogic() {
+		return findReplaceLogic;
+	}
+
+	@Override
+	public void performReplaceAll() {
+		openReplaceDialog();
+		replaceAllButton.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void performReplace() {
+		openReplaceDialog();
+		replaceButton.notifyListeners(SWT.Selection, null);
+	}
+
+	public boolean isReplaceDialogOpen() {
+		return dialogAccessor.getBoolean("replaceBarOpen");
+	}
+
+	public void openReplaceDialog() {
+		if (!isReplaceDialogOpen() && Objects.nonNull(openReplaceDialog)) {
+			openReplaceDialog.notifyListeners(SWT.Selection, null);
+			replace= (Text) dialogAccessor.get("replaceBar");
+			replaceButton= (ToolItem) dialogAccessor.get("replaceButton");
+			replaceAllButton= (ToolItem) dialogAccessor.get("replaceAllButton");
+		}
+	}
+
+	public void closeReplaceDialog() {
+		if (isReplaceDialogOpen() && Objects.nonNull(openReplaceDialog)) {
+			openReplaceDialog.notifyListeners(SWT.Selection, null);
+			replace= null;
+			replaceButton= null;
+			replaceAllButton= null;
+		}
+	}
+
+	@Override
+	public void performReplaceAndFind() {
+		performReplace();
+	}
+
+	@Override
+	public void assertInitialConfiguration() {
+		assertUnselected(SearchOptions.REGEX);
+		assertUnselected(SearchOptions.WHOLE_WORD);
+		assertUnselected(SearchOptions.CASE_SENSITIVE);
+		if (!doesTextViewerHaveMultiLineSelection(findReplaceLogic.getTarget())) {
+			assertSelected(SearchOptions.GLOBAL);
+			assertTrue(findReplaceLogic.isActive(SearchOptions.GLOBAL));
+		} else {
+			assertUnselected(SearchOptions.GLOBAL);
+			assertFalse(findReplaceLogic.isActive(SearchOptions.GLOBAL));
+		}
+	}
```

Comment:
This is a follow-up to https://github.com/eclipse-platform/eclipse.platform.ui/pull/1192#discussion_r1598370379 and the changes made to address it: you explained that no test has been failing even though the behavior is broken, so probably there are tests or assertions missing or wrong. I see a particular problem in this initial configuration check, which is rather incompletely (also compared to the one of `DialogAccess`). There are no enablement checks, and the `assertEnabled` and `assertDisabled` methods are not even called.

Reply:
I agree that enablement states should be checked for as well - now that we have assertEnabled and assertDisabled moved into this class, I could very easily add the corresponding checks.

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/2f9136a25069244769073ee77ad5be15b5dadeee